### PR TITLE
Feature/tmarn 199/nl puff graphql integration

### PR DESCRIPTION
**Pull request by skumar-newsuk**
#### Description
Graphql integration is done with the new apollo/client as the previous one (react-apollo) is deprecated. Verified the query and updated the newsletter status successfully on both Android and iOS.

#### Checklist
 - [x] Unit tests
 - [x] Tested on iOS simulator
 - [x] Tested on Android emulator
 - [x] Tested on Tablet

[TMARN-199]: https://nidigitalsolutions.jira.com/browse/TMARN-199?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,6 +75,7 @@
   },
   "homepage": "https://github.com/newsuk/times-components-native#readme",
   "dependencies": {
+    "@apollo/client": "^3.7.14",
```
Comment (petermason) [inline package.json:78]: can you remove the ^ to fix the version. This can cause issues on CI

File: package.json [hunk 2]
```diff
@@ -84,6 +85,7 @@
     "apollo-link": "1.2.4",
     "apollo-link-http": "1.5.14",
     "apollo-utilities": "1.2.1",
+    "cross-fetch": "^3.1.6",
```
Comment (petermason) [inline package.json:88]: please remove the ^ and any others

File: packages/article-skeleton/src/article-body/newsletterpuff/index.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+  // Get Newsletter to see if the user is already subscribed
+  const { loading, error, data } = useQuery(GET_NEWSLETTER, {
+    variables: { code },
+  });
+
+  useEffect(() => {
+    if (loading) {
+      console.log("Loading...");
+    }
```
Comment (petermason) [inline packages/article-skeleton/src/article-body/newsletterpuff/index.tsx:54]: if not needed if only for console log

File: packages/article-skeleton/src/article-body/newsletterpuff/index.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    if (error) {
+      console.error(error);
+    }
```
Comment (petermason) [inline packages/article-skeleton/src/article-body/newsletterpuff/index.tsx:57]: if not needed if only for console log

File: packages/article-skeleton/src/article-body/newsletterpuff/index.tsx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    if (data) {
+      setIsSubscribed(data.newsletter.isSubscribed);
+    }
+  }, [loading, error, data, justSubscribed]);
+
+  // Mutation returns callback function to use
+  const [subscribeNewsletter] = useMutation(SUBSCRIBE_NEWSLETTER, {
+    onError: error => {
+      console.log("subscribeNewsletter", error);
```
Comment (petermason) [inline packages/article-skeleton/src/article-body/newsletterpuff/index.tsx:66]: should we handle the erro correctly?

File: packages/article-skeleton/src/article-body/newsletterpuff/index.tsx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+  const onClickSubscribe = () => {
+    if (!justSubscribed) {
+      subscribeNewsletter({ variables: { code } });
+    } 
+  };
+
+  return (
+    <View>
+      {!isSubscribed && (
```
Comment (petermason) [inline packages/article-skeleton/src/article-body/newsletterpuff/index.tsx:81]: It's better to handle the conditional logic in a component

File: packages/article-skeleton/src/styles/newsletterpuff/index.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            font: "body",
+            fontSize: "bodyMobile",
+          }),
+        fontFamily: 'TimesDigitalW04-Regular',
+        paddingHorizontal:16,
+        color:theme.colors.nlCommonText,
+
+    },
+    titleText: {
+        fontWeight: "700",
```
Comment (petermason) [inline packages/article-skeleton/src/styles/newsletterpuff/index.ts:37]: not required, font weight provided by the font

File: packages/article-skeleton/src/styles/newsletterpuff/index.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        borderWidth: 1,
+        borderColor: 'black',
+        backgroundColor: 'transparent',
+        padding: 10,
+        marginVertical:10,
+        marginHorizontal:16,
+      },
+      buttonText: {
+        color: '#333333',
+        fontWeight: "500",
```
Comment (petermason) [inline packages/article-skeleton/src/styles/newsletterpuff/index.ts:53]: font weight provided by the font

File: packages/article-skeleton/src/styles/newsletterpuff/index.ts [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    button: {
+        borderWidth: 1,
+        borderColor: 'black',
+        backgroundColor: 'transparent',
+        padding: 10,
+        marginVertical:10,
+        marginHorizontal:16,
+      },
+      buttonText: {
+        color: '#333333',
```
Comment (petermason) [inline packages/article-skeleton/src/styles/newsletterpuff/index.ts:52]: is there a theme for this. no hardcoded values for colours

File: packages/article-skeleton/src/article-body/newsletterpuff/index.tsx (hunk 6 above)
Comment (petermason) [reply packages/article-skeleton/src/article-body/newsletterpuff/index.tsx:86]: Maybe create components that handle the render logic.

**Comment by skumar-newsuk** [comment]
Addressed all the review comments. Kindly re-review my code when ever you get time.

**Pull request merged**